clang/test/SemaSYCL/intel-fpga-uses-global-work-offset.cpp

```diff
@@ -0,0 +1,49 @@
+// RUN: %clang_cc1 -Wno-return-type -fsycl-is-device -fcxx-exceptions -fsyntax-only -ast-dump -verify -pedantic %s | FileCheck %s
+
+struct FuncObj {
+  [[intelfpga::uses_global_work_offset(1)]] void operator()() {}
+};
+
+template <typename name, typename Func>
+void kernel(Func kernelFunc) {
+  kernelFunc();
+}
+
+int main() {
+  // CHECK: SYCLIntelUsesGlobalWorkOffsetAttr{{.*}}Enabled
+  kernel<class test_kernel1>([]() {
+    FuncObj();
+  });
+
+  // CHECK: SYCLIntelUsesGlobalWorkOffsetAttr
+  // CHECK-NOT: Enabled
+  kernel<class test_kernel2>(
+      []() [[intelfpga::uses_global_work_offset(0)]]{});
+
+  // CHECK: SYCLIntelUsesGlobalWorkOffsetAttr{{.*}}Enabled
+  // expected-warning@+2{{attribute should be 0 or 1. Adjusted to 1}}
```

Comment:
Just to clarify, is it OK to leave the attribute's naming unspecified in the checks?

Reply:
As long as there are no other attributes here, I don't see any problem in shortening warning/error message. It's even simpler to read the test

Reply:
If there are two attributes incorrectly used on the same lambda - we will be given errors/warnings pointing on the same string. I'd prefer to see attr's name to simplify distinguishing.